### ISSUE TITLE
Access Denied with mix zig.get - M1 MacOS

### ISSUE THREAD

**Issue by clayscode** [E-xyza/zigler#455]
Hi,

Trying to install but getting an access denied with the `mix zig.get` command. I'm running this on an M1

```(venv) clay@Cmacbook data % mix zig.get
** (MatchError) no match of right hand side value: {:ok, {{~c"HTTP/1.1", 403, ~c"Forbidden"}, [{~c"connection", ~c"keep-alive"}, {~c"date", ~c"Fri, 17 May 2024 14:00:19 GMT"}, {~c"via", ~c"1.1 4b8be55299e599fba7c9de026a1b5eca.cloudfront.net (CloudFront)"}, {~c"server", ~c"AmazonS3"}, {~c"content-length", ~c"303"}, {~c"content-type", ~c"text/html; charset=utf-8"}, {~c"x-cache", ~c"Error from cloudfront"}, {~c"x-amz-cf-pop", ~c"ORD58-P7"}, {~c"x-amz-cf-id", ~c"-dHQlWv-Jj54gqkkJs8O6Kj0j3SphlnU-xNCcPQTTxjeRh-8aV1KWw=="}], "<html>\n<head><title>403 Forbidden</title></head>\n<body>\n<h1>403 Forbidden</h1>\n<ul>\n<li>Code: AccessDenied</li>\n<li>Message: Access Denied</li>\n<li>RequestId: AKXBV2JYE36CN8ER</li>\n<li>HostId: FD2kWSkGlzMNreMhyKacKrUyaw00bMvyTllaWKJmWMmT4t/3J2YkjCZ1YXfolTq8bMZZamopCNA=</li>\n</ul>\n<hr/>\n</body>\n</html>\n"}}
    mix.tasks/zig.get.ex:231: Mix.Tasks.Zig.Get.http_get!/1
    mix.tasks/zig.get.ex:254: Mix.Tasks.Zig.Get.fetch_signature!/1
    mix.tasks/zig.get.ex:54: Mix.Tasks.Zig.Get.run/1
    (mix 1.16.1) lib/mix/task.ex:478: anonymous fn/3 in Mix.Task.run_task/5
    (mix 1.16.1) lib/mix/cli.ex:96: Mix.CLI.run_task/2
    /opt/homebrew/bin/mix:2: (file)
```

**Comment by clayscode** [comment]
If I install zig via homebrew and then create a symlink, I get a little closer to this working:

```
mkdir -p /Users/clay/Library/Caches/mix/installs/elixir-1.15.7-erts-14.1.1/2b66fb1edcb76a2a6e18cb03cfb2340b/deps/zigler/zig/zig-macos-aarch64-0.11.0/
ln -s /opt/homebrew/bin/zig /Users/clay/Library/Caches/mix/installs/elixir-1.15.7-erts-14.1.1/2b66fb1edcb76a2a6e18cb03cfb2340b/deps/zigler/zig/zig-macos-aarch64-0.11.0/zig
```

However when I compile the example module I get the following error:
<img width="1005" alt="Screen Shot 2024-05-17 at 9 13 51 AM" src="https://github.com/E-xyza/zigler/assets/16871737/8d0a7ff3-0daf-41e1-8e11-a53946abb113">

**Comment by ityonemo** [comment]
sorry, this is a regression, you can fix it by pinning the zig_get dependency:

`{:zig_get, "~> 0.11.3", runtime: false, override: true}`

**Comment by ityonemo** [comment]
aha, I see, are you doing this in a livebook?

**Comment by tracy-o** [comment]
Getting the same error on MacOS with M1. Not using homebrew (downloaded zig from the official page to `~/zig`).

Running `mix zig.get` I keep getting:
```shell
** (MatchError) no match of right hand side value: {:ok, {{~c"HTTP/1.1", 403, ~c"Forbidden"}, [{~c"connection", ~c"keep-alive"}, {~c"date", ~c"Fri, 05 Jul 2024 12:12:52 GMT"}, {~c"via", ~c"1.1 942cb3fbe68b5c10602773b4d1b1c0e8.cloudfront.net (CloudFront)"}, {~c"server", ~c"AmazonS3"}, {~c"content-length", ~c"303"}, {~c"content-type", ~c"text/html; charset=utf-8"}, {~c"x-cache", ~c"Error from cloudfront"}, {~c"x-amz-cf-pop", ~c"LHR61-P3"}, {~c"x-amz-cf-id", ~c"He18ETnF4IKZ6zm2QhxgYPaf2VfgMrGZgn1GHo-g4tofsbVFLaefqA=="}], "<html>\n<head><title>403 Forbidden</title></head>\n<body>\n<h1>403 Forbidden</h1>\n<ul>\n<li>Code: AccessDenied</li>\n<li>Message: Access Denied</li>\n<li>RequestId: K9GAZ860JBXRE6H0</li>\n<li>HostId: 80xoeLJxk6C716eDHAWk0A5anqspbYWaqTZFR4+sq16gHFMIHmmQbgbG8ha6e4Yh6sdFML5XNY4=</li>\n</ul>\n<hr/>\n</body>\n</html>\n"}}
    (zig_get 0.11.3) mix.tasks/zig.get.ex:231: Mix.Tasks.Zig.Get.http_get!/1
    (zig_get 0.11.3) mix.tasks/zig.get.ex:254: Mix.Tasks.Zig.Get.fetch_signature!/1
    (zig_get 0.11.3) mix.tasks/zig.get.ex:54: Mix.Tasks.Zig.Get.run/1
    (mix 1.16.2) lib/mix/task.ex:478: anonymous fn/3 in Mix.Task.run_task/5
    (mix 1.16.2) lib/mix/cli.ex:96: Mix.CLI.run_task/2
    ~/.asdf/installs/elixir/1.16.2-otp-26/bin/mix:2: (file)
```

**Comment by luechtdev** [comment]
`mix zig.get --os macos` worked for me (M3 Mac) since zig_get assumed 'os=darwin' but zig expected 'os=macos'.
Alternatively you can try to insert a debugging statement/breakpoint in `deps/zig_get/mix.tasks/zi.get:231` and have a look at the url. Most likely error will be that the url does not exist on the zig download server.

**Comment by tracy-o** [comment]
Setting `--os macos` worked for me! Thank you @luechtdev

**Comment by ityonemo** [comment]
fixed in 0.12.1
